### DISCUSSION TITLE
Is there a way to check if client connection is OK?

### DISCUSSION THREAD

**Discussion by Sulomus** [supabase/supabase#5797] [Questions]
- I am building an IoT devices that will communicate with Supabase.
- The IoT things will run `linux` environment with NODE.js and Supabase client libraries as a way to communicate with Cloud.

I need to create a mechanism that will check if the connection to the Supabase is OK and if not then reconnect the device (just create a new client maybe?).

By "supabase client" I mean an object that `supabase.createClient` function returns:
`const supabaseClient = createClient(supabaseUrl, supabaseKey);`

Is there some function or property like `supabaseClient.connected()` that I can use to check this?

**Comment by GaryAustin1** [comment]
What does it mean to be “connected”?

The API requests are all pretty much stateless and you need to check for error on them for things like no internet connection when you execute them. 

If you mean signed with valid access token, that is more monitoring auth state change and there is an auth function for that. 

I use one of my realtime connections to provide an online indicator to the user by monitoring status being connected.  

.subscribe((status)=> {
                if (status === "SUBSCRIBED”) doOnline() else doOffline()}
 
 But this is still to a different port on the Supabase server than the API and is polled in a websocket with a one minute timeout.  This means you still have handle API request connection errors

**Reply by ThomasMeinzGroup** [reply]
it means the anon key and the url are checked for validity.
no user credentials available.
